app/dashboard/components/charts/OnsightPercentage.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+
+import React from "react";
+import RawDataRow, { RawDataList } from "../../types/raw-data-from-mountain-project";
+import * as d3 from 'd3';
+import { useEffect, useRef } from "react";
+
+// We can define what this object will look like after we decide exactly what we want to pass in
+interface Props {
+    data: any
+}
+
+const OnsightPercentage: React.FC<Props> = ({ data }: Props) => {
+  const sportStyleCombinations = ["Sport", "Sport, Alpine", "Trad", "TR", "Sport, TR"];
+  const onsightLeadStyle = ["Onsight", "Flash"];
+  const justSport = data.filter((oneRoute: RawDataRow) => sportStyleCombinations.includes(oneRoute["Route Type"]));
+  const onsightSport = justSport.filter((oneRoute: RawDataRow) => onsightLeadStyle.includes(oneRoute["Lead Style"]));
+  const groupAllSportByGrade = d3.group(justSport, (oneRoute: RawDataRow) => oneRoute.Rating);
+  console.log(groupAllSportByGrade);
+  const margin = ({top: 20, right: 0, bottom: 30, left: 40}), width = 460 - margin.left - margin.right, height = 400 - margin.top - margin.bottom;
+  const x = d3.scaleBand().domain(justSport.map((oneRoute: RawDataRow) => oneRoute.Rating)).range([margin.left, width - margin.right]).padding(0.1);
+  const xAxis = d3.axisBottom(x).tickSizeOuter(0);
+  const y = d3.scaleLinear().domain(justSport.map((oneRoute: RawDataRow) => oneRoute.Rating)).nice().range([height - margin.bottom, margin.top]);
+  const yAxis = d3.axisLeft(y).ticks(null, "%")
```

Comment:
@lukehtravis It looks like these miiiiight belong in utils.tsx? So that [these statements](https://github.com/lukehtravis/climbing-data-dashboard/blob/b109c64158f2d4522f50bd52a4db368f7aeb319b/app/dashboard/components/Visualizations.tsx#L13) can leverage the utility function,